components/player-view.js

```diff
@@ -58,10 +58,17 @@ module.exports = React.createClass({
 
     requestAnimationFrame(this.step);
   },
-  componentWillMount: function() {
+  componentWillUpdate: function() {
     this.isYt = !!~this.props.domain.indexOf('youtube.com');
   },
-  componentDidMount: function() {
+  componentDidUpdate: function(prevProps, prevState) {
+    if (prevProps.src !== this.props.src) {
```

Comment:
Will this line try to `attachVideoListeners` if src is reset to an empty string?